src/test/java/com/palantir/common/streams/BufferingSpliteratorTests.java

```diff
@@ -34,7 +35,10 @@
 import org.mockito.junit.MockitoJUnitRunner;
 
 @RunWith(MockitoJUnitRunner.class)
-public final class InCompletionOrderSpliteratorTests {
+public final class BufferingSpliteratorTests {
+    // None of the tests in this class should be affected by the choice of completion strategy.
```

Comment:
Why not use a parameterized runner then?

Reply:
Also, we really should have tests that distinguish the two cases

Reply:
We do have such tests - they're in MoreFuturesTests.

I'll do the parameterization.